lib/verdict/storage/redis_storage.rb

```diff
@@ -6,33 +6,56 @@ class RedisStorage < BaseStorage
       attr_accessor :redis, :key_prefix
 
       def initialize(redis = nil, options = {})
-        @redis = redis
+        if !redis.nil? && !redis.respond_to?(:with)
+          @redis = ConnectionPoolLike.new(redis)
+        else
+          @redis = redis
+        end
+
         @key_prefix = options[:key_prefix] || 'experiments/'
       end
 
       protected
 
+      class ConnectionPoolLike
+        def initialize(redis)
+          @redis = redis
+        end
+
+        def with
+          yield @redis
+        end
+      end
+
       def get(scope, key)
-        redis.hget(scope_key(scope), key)
+        redis.with do |r|
+          r.hget(scope_key(scope), key)
+        end
       rescue ::Redis::BaseError => e
         raise Verdict::StorageError, "Redis error: #{e.message}"
       end
 
       def set(scope, key, value)
-        redis.hset(scope_key(scope), key, value)
+        redis.with do |r|
```

Comment:
instead of `r` I would use `conn`. So it would be more clear what's the difference between redis and this object.